test-data/unit/check-generics.test

```diff
@@ -1505,7 +1505,7 @@ T = TypeVar('T')
 class C(Generic[T]):
     def __init__(self) -> None: pass
 x = C # type: Callable[[], C[int]]
-y = C # type: Callable[[], int] # E: Incompatible types in assignment (expression has type C[T], variable has type Callable[[], int])
+y = C # type: Callable[[], int] # E: Incompatible types in assignment (expression has type Type[C[T]], variable has type Callable[[], int])
```

Comment:
> I am also not entirely sure I understand what's wrong with ``Type[C[T]]``

PEP 484 specifies that bare (unsubscribed) type is equivalent to ``C[Any]``. Moreover, writing ``C[T]`` is invalid in this context (and in other examples I mentioned below) and will trigger a mypy error ``Invalid type '__main__.T'``.

Does fixing this actually requires extensive code duplication? If yes, then maybe it is better to import ``erasetype``.